### PR TITLE
[FIX] do not delete xml_id for routing sequence in mrp module as it will be renamed installing mrp_routing

### PR DESCRIPTION
**Pull request by sergiocorato**
Ref https://github.com/OCA/manufacture/pull/1102

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Please put module name in the commit message, and we are tagging them as `[OU-FIX]` for differentiating them from regular fixes in for example the OpenUpgrade framework. Can you give some links to support the non removal?

File: openupgrade_scripts/scripts/mrp/14.0.2.0/post-migration.py [hunk 1]
```diff
@@ -89,6 +89,4 @@ def migrate(env, version):
     map_mrp_production_state_planned(env)
     generate_workorders_for_draft_orders(env)
     openupgrade.load_data(env.cr, "mrp", "14.0.2.0/noupdate_changes.xml")
-    openupgrade.delete_records_safely_by_xml_id(
-        env, ["mrp.mrp_routing_rule", "mrp.sequence_mrp_route"]
-    )
+    openupgrade.delete_records_safely_by_xml_id(env, ["mrp.mrp_routing_rule"])
```
Comment (MiquelRForgeFlow) [inline openupgrade_scripts/scripts/mrp/14.0.2.0/post-migration.py:92]:
```suggestion
    openupgrade.delete_records_safely_by_xml_id(env, ["mrp.mrp_routing_rule"])
    # not deleting mrp.sequence_mrp_route, as it is used in OCA/manufacture mrp_routing module.
```

Comment (MiquelRForgeFlow) [inline openupgrade_scripts/scripts/mrp/14.0.2.0/post-migration.py:92]:
```suggestion
    openupgrade.delete_records_safely_by_xml_id(env, ["mrp.mrp_routing_rule"])
    if not openupgrade.is_module_installed(cr.cr, "mrp_routing"):
        openupgrade.delete_records_safely_by_xml_id(env, ["mrp.sequence_mrp_route"])
```

**Review by MiquelRForgeFlow — CHANGES_REQUESTED** [review]

File: openupgrade_scripts/scripts/mrp/14.0.2.0/post-migration.py (hunk 1 above)
Comment (sergiocorato) [reply openupgrade_scripts/scripts/mrp/14.0.2.0/post-migration.py:92]: Thanks @MiquelRForgeFlow, but `mrp_routing` is a brand new module in v. 14.0, so this check will always result true and delete the sequence.

**Comment by pedrobaeza** [comment]
I don't think this should be changed, as only installing the other module, you will need this record, so this shouldn't be unconditionally not deleted.

File: openupgrade_scripts/scripts/mrp/14.0.2.0/post-migration.py (hunk 1 above)
Comment (MiquelRForgeFlow) [reply openupgrade_scripts/scripts/mrp/14.0.2.0/post-migration.py:92]: But when installing `mrp_routing`, the migration script of `mrp_routing` will not execute 🤔

Comment (sergiocorato) [reply openupgrade_scripts/scripts/mrp/14.0.2.0/post-migration.py:92]: Mmm, good point, I'll correct that.

**Comment by sergiocorato** [comment]
> I don't think this should be changed, as only installing the other module, you will need this record, so this shouldn't be unconditionally not deleted.

I agree, this will only be useful in those migrations where this module will be installed.

**Comment by pedrobaeza** [comment]
But is there a reason to keep this one? A new one with the exact same parameters is not enough?

**Comment by sergiocorato** [comment]
A new one will not preserve current sequence and possible customization.
Anyway the pre_init_hook I added here https://github.com/OCA/manufacture/pull/1102 doesn't preserve them too, I don't know why.

File: openupgrade_scripts/scripts/mrp/14.0.2.0/post-migration.py (hunk 1 above)
Comment (pedrobaeza) [reply openupgrade_scripts/scripts/mrp/14.0.2.0/post-migration.py:92]: Migration scripts for newly installed modules happen in the context of OpenUpgrade (if you load `openupgrade_framework`. Please put here the condition to find the module in the module list, which is likely that you are going to install the module.

Comment (sergiocorato) [reply openupgrade_scripts/scripts/mrp/14.0.2.0/post-migration.py:92]: So `pre-migration.py` script it's called even in a newly installed module?

I don't know what the phrase `put here the condition to find the module in the module list` means, does it refer to the `openupgrade.is_module_installed` method?

Comment (pedrobaeza) [reply openupgrade_scripts/scripts/mrp/14.0.2.0/post-migration.py:92]: Yes and I mean:

`if env["ir.module.module"].search([("name", "=", "mrp_routing")]):`

Comment (sergiocorato) [reply openupgrade_scripts/scripts/mrp/14.0.2.0/post-migration.py:92]: Wow, I'll try, thanks.

**Comment by sergiocorato** [comment]
I haven't found a way to make it working, so I'll do it manually.

**Pull request closed without merging**